### PR TITLE
Wfe 29

### PR DESCRIPTION
**Pull request by joehen**
added some test cases for the mdsd blackboard and qvto to confirm WFE-29 but the issue seems to be solved as QVTOTestcase seems to work with non existing output partition resources.

### CODE REVIEW

File: tests/de.uka.ipd.sdq.workflow.mdsd.tests/src/de/uka/ipd/sdq/workflow/mdsd/tests/MDSDBlackboardTests.java [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+import org.eclipse.emf.ecore.xmi.impl.XMIResourceFactoryImpl;
+
+
+public class MDSDBlackboardTests {
+	
+	MDSDBlackboard blackboard;
+	
+	@BeforeClass
+	public static void registerFactories() {
+		Resource.Factory.Registry reg = Resource.Factory.Registry.INSTANCE;
```
Comment (seiferma) [inline tests/de.uka.ipd.sdq.workflow.mdsd.tests/src/de/uka/ipd/sdq/workflow/mdsd/tests/MDSDBlackboardTests.java]: Please fix the indentation (in the whole file)

File: tests/de.uka.ipd.sdq.workflow.mdsd.tests/src/de/uka/ipd/sdq/workflow/mdsd/tests/QVTOTests.java [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+	public void setUp() {
+		this.blackboard = new MDSDBlackboard();
+	}
+
+	@Test
+	public void testQVTOTransformationJob() throws JobFailedException, UserCanceledException, IOException {
+
+		ModelLocation[] inoutLocations = new ModelLocation[2];
+		
+		URI inputModelURI = URI.createPlatformPluginURI("/de.uka.ipd.sdq.workflow.mdsd.tests//models/test.ecore", true);
```
Comment (seiferma) [inline tests/de.uka.ipd.sdq.workflow.mdsd.tests/src/de/uka/ipd/sdq/workflow/mdsd/tests/QVTOTests.java]: The double slahes after the project name are not necessary.

File: tests/de.uka.ipd.sdq.workflow.mdsd.tests/src/de/uka/ipd/sdq/workflow/mdsd/tests/QVTOTests.java [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+		URI outputModelURI = URI.createPlatformPluginURI("/de.uka.ipd.sdq.workflow.mdsd.tests//models/output.ecore", true);
+		URI scriptFileURI = URI.createPlatformPluginURI("/de.uka.ipd.sdq.workflow.mdsd.tests//models/test.qvto", true);
```
Comment (seiferma) [inline tests/de.uka.ipd.sdq.workflow.mdsd.tests/src/de/uka/ipd/sdq/workflow/mdsd/tests/QVTOTests.java]: Would be nice to support plugin and standalone tests. File URIs work in general but the transformation script is not loaded properly. The easiest solution is to create a conditional assignment based on org.eclipse.core.runtime.Platform.isRunning(). The other URIs can be changed to file URIs as used in MDSDBlackboardTests.

File: tests/de.uka.ipd.sdq.workflow.mdsd.tests/src/de/uka/ipd/sdq/workflow/mdsd/tests/MDSDBlackboardTests.java [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+    	
+		SavePartitionToDiskJob job = new SavePartitionToDiskJob("testPartitionId");
+		job.setBlackboard(blackboard);
+		
+		final NullProgressMonitor monitor = new NullProgressMonitor();
+		job.execute(monitor);
+		
+		Assert.assertTrue(partition.wasSaved());
+		
+
```
Comment (seiferma) [inline tests/de.uka.ipd.sdq.workflow.mdsd.tests/src/de/uka/ipd/sdq/workflow/mdsd/tests/MDSDBlackboardTests.java]: Minor: You could reduce the blank lines here.

File: tests/de.uka.ipd.sdq.workflow.mdsd.tests/src/de/uka/ipd/sdq/workflow/mdsd/tests/MDSDBlackboardTests.java (hunk 1 above)
Comment (joehen) [reply tests/de.uka.ipd.sdq.workflow.mdsd.tests/src/de/uka/ipd/sdq/workflow/mdsd/tests/MDSDBlackboardTests.java]: new Eclipse --> wrong formatter

**Pull request merged**